invenio_records_resources/services/files/tasks.py

```diff
@@ -39,3 +45,57 @@ def fetch_file(service_id, record_id, file_key):
 
     except FileKeyNotFoundError as e:
         current_app.logger.error(e)
+
+
+# TODO update siegfried signatures (`sf -update`) regularly
+@shared_task(ignore_result=True)
+def detect_file_type(bucket_id, file_key):
+    """Detect the format of the file using siegfried."""
+    # TODO maybe we should go through the Records-Resources files API instead?
+    ov = ObjectVersion.get(bucket_id, file_key)
+    if ov.file is None:
+        return
+
+    # TODO the original filename is lost (renamed to 'data'), but sf uses the filename
+    #      for parts of its algorithm; possible solutions:
+    #      * create a temporary alias (link?) to the file and pass that to sf
+    #      * pipe the file's contents into sf via stdin and use the `-name` arg
+
+    # TODO question: could we utilize siegfried's server mode?
+
+    mimetype, pronom_id = None, None
+    try:
+        sf_bin = "sf"
+        # TODO this may only be possible for 'local' storage?
+        sf_output = sp.check_output([sf_bin, "-json", ov.file.uri], text=True)
+        result = json.loads(sf_output)
+
+        for file_info in result.get("files", []):
+            # only consider results for the file in question
+            if file_info.get("filename") != ov.file.uri:
+                continue
+
+            if not file_info.get("errors", None) and file_info.get("matches", []):
+                for match in file_info["matches"]:
+                    if match["ns"] == "pronom":
+                        pronom_id = match["id"]
+
+                        # NOTE: there may be results other than for the "pronom" ns
+                        #       which may actually deliver better matches
+                        #       e.g. for the `sway-vulkan` script, the sf website
+                        #           (https://www.itforarchivists.com/siegfried)
+                        #           reports "plain text file" and no mimetype for PRONOM
+                        #           but "shell script" (and a mimetype) for the
+                        #           "freedesktop.org" ns
+                        if match["mime"]:
+                            mimetype = match["mime"]
+
+    except Exception as e:
+        print(e)
+
+    if mimetype is not None:
+        ov.mimetype = mimetype
+    if pronom_id is not None:
+        ObjectVersionTag.create_or_update(ov, "pronom_id", pronom_id)
```

Comment:
should this be stored in another location?
maybe in the file "metadata" which is reported in the REST API (e.g. dimensions of images)?